ProteinsWebTeam/interpro7-client
# changing offline-plugin for workbox

The logic is very similar to what we had with  `offline-plugin`. 
The service worker gets registered in the `App` component, and the `offline.js` checks every 30min if there is a new `sw.module.js`in which case prompts the message to update, and after user accepts, the browser gets reloaded to fully activate the new service worker.

**To test this functionality**
The service worker is only on operation from a bundled app(not form dev server). Therefore the steps I used to test the service worker are:
1. Clean any previous deployment: `npm run clean`
2. Remove `"/interpro"` from website `config.yml`:  `website: //localhost:8000/`
3. Build a deployment:  Use `npm run bundle` to skip the tests.
4. Run a web server for the `/dist` folder: `npx http-server dist`
5. Open the website in the browser you are testing. You can check the Application tab in the developer tools to make sure the service worker running.
6. Shutdown the server `^c`.
7. if you are in the home page, you can refresh and the page will still load, even if the server is off.
8. Make a change in the code, something you can make sure the new version is running.
9. repeat steps 1 to 4 to build the new version.
10.  The service worker is configure to check for a new version every half an hour so in 30 min the "New version" notification should appear. and once accepted the website should refresh using the new version.

*Note:* You can check the time to check for a new version in `stc/offline.js` in the constant `DELAY_BEFORE_CHECKING_NEW_VERSION`. but you need to do this in the SW of the first built, as that's the one running the timer.
